src/protocol/libp2p/kademlia/record.rs

```diff
@@ -118,3 +121,34 @@ pub struct PeerRecord {
     /// The provided record.
     pub record: Record,
 }
+
+/// A record keeping information about a content provider.
+#[derive(Clone, Debug, Eq, PartialEq, Hash)]
+pub struct ProviderRecord {
+    /// Key of the record.
+    pub key: Key,
+
+    /// Key of the provider, based on its peer ID.
+    pub provider: PeerId,
+
+    /// Cached addresses of the provider.
+    pub addresses: Vec<Multiaddr>,
+
+    /// The expiration time of the record. The provider records must always have the expiration
+    /// time.
+    pub expires: Instant,
```

Comment:
Nice! I wonder if we should do a similar thing for the `Record`. It may not be entirely compliant with the spec, however it will force records out of the store. Do you think it would be beneficial to cap `Record::expires` at say 24h (or other number)?

Reply:
Not sure about `Record`. It's specifically highlighted in the libp2p protocol spec that provider records must expire, because providers can go offline.

Reply:
Aah, I see what you mean. Yes, we can introduce some maximum TTL for records and require publishers to renew them, to not store stale records forever. Let's think if there are applications for long-lived records where the original publisher goes offline. I.e., we can run a command once and publish a record — are there scenarios where this can be useful?

Reply:
May be some credential storing, or a DNS-like system?